File: Backend/src/main/java/app/alertify/controller/dto/summary/v1/Check.java
package app.alertify.controller.dto.summary.v1;

import java.time.Duration;
import java.util.Date;
import java.util.List;
import java.util.Objects;

public class Check {

    private String name;
    private Date lastSuccess;
    private Duration period;
    private List<CheckError> errors;
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public Date getLastSuccess() {
		return lastSuccess;
	}
	public void setLastSuccess(Date lastSuccess) {
		this.lastSuccess = lastSuccess;
	}
	public Duration getPeriod() {
		return period;
	}
	public void setPeriod(Duration period) {
		this.period = period;
	}
	public List<CheckError> getErrors() {
		return errors;
	}
	public void setErrors(List<CheckError> errors) {
		this.errors = errors;
	}
	@Override
	public int hashCode() {
		return Objects.hash(errors, lastSuccess, name, period);
	}
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Check other = (Check) obj;
		return Objects.equals(errors, other.errors) && Objects.equals(lastSuccess, other.lastSuccess)
				&& Objects.equals(name, other.name) && Objects.equals(period, other.period);
	}
	@Override
	public String toString() {
		return "Check [name=" + name + ", lastSuccess=" + lastSuccess + ", period=" + period + ", errors=" + errors
				+ "]";
	}

	
}

File: Frontend/src/app/data/check.ts
import { ErrorCheck } from "./error-check";
import { Status } from "./status";

export class Check {
    private _name: string;
    private _status: Status;
    private _lastSuccess: Date;
    private _period: Number;
    private _errors: ErrorCheck[];

    constructor(name: string, status: Status, lastSuccess: Date, period: Number, errors: ErrorCheck[]) {
        this._name = name;
        this._status = status;
        this._lastSuccess = lastSuccess;
        this._period = period;
        this._errors = errors;
    }

    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }
    public get status(): Status {
        return this._status;
    }
    public set status(value: Status) {
        this._status = value;
    }
    public get lastSuccess(): Date {
        return this._lastSuccess;
    }
    public set lastSuccess(value: Date) {
        this._lastSuccess = value;
    }
    public get period(): Number {
        return this._period;
    }
    public set period(value: Number) {
        this._period = value;
    }
    public get errors(): ErrorCheck[] {
        return this._errors;
    }
    public set errors(value: ErrorCheck[]) {
        this._errors = value;
    }
    
}
